### PR TITLE
operator helm chart: helm or user creates the release namespace

### PR DESCRIPTION
**Pull request by papdaniel**
Namespace creation by helm template is not a good practice. It should be created by the user or Helm itself
See the [issue](https://github.com/istio/istio/issues/30733).

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by shamsher31** [comment]
Not stale

**Comment by shamsher31** [comment]
@papdaniel Please sign the CLA.

**Comment by jacob-delgado** [comment]
This LGTM. Having helm manage the namespace isn't proper behavior. Let me know when the integration tests are passing @papdaniel and you've signed the CLA.

**Comment by shamsher31** [comment]
@papdaniel Do you want to take it further to help this PR merge?

**Comment by shamsher31** [comment]
@papdaniel Just run `make gen` and `make lint` to fix lint issues.

**Comment by papdaniel** [comment]
@shamsher31, everything seems good

File: operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml [hunk 1]
```diff
@@ -6068,6 +6071,10 @@ spec:
             value: istio-ingressgateway
           - name: ISTIO_META_OWNER
             value: kubernetes://apis/apps/v1/namespaces/istio-system/deployments/istio-ingressgateway
+          - name: ISTIO_META_MESH_ID
```
Comment (shamsher31) [inline operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml:6074]: Is this change related to this PR?

Comment (papdaniel) [reply operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml:6074]: Nope. I don't know what is this, and port numbers neither.

**Comment by papdaniel** [comment]
@shamsher31 @costinm @sdake @ostromart is there any reason you don't merge it? It's been waiting for weeks.

**Comment by jacob-delgado** [comment]
> @shamsher31 @costinm @sdake @ostromart is there any reason you don't merge it? It's been waiting for weeks.

If you get it rebased I'll work with the appropriate people to get it merged. Let me know when it's ready.

**Comment by papdaniel** [comment]
> > @shamsher31 @costinm @sdake @ostromart is there any reason you don't merge it? It's been waiting for weeks.
> 
> If you get it rebased I'll work with the appropriate people to get it merged. Let me know when it's ready.

Thanks @jacob-delgado, it's ready to merge.

**Review by shamsher31 — APPROVED** [review]
Thanks

File: operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml [hunk 2]
```diff
@@ -5774,6 +5774,9 @@ rules:
   - apiGroups: ["authorization.k8s.io"]
     resources: ["subjectaccessreviews"]
     verbs: ["create"]
+  - apiGroups: ["multicluster.x-k8s.io"]
```
Comment (shamsher31) [inline operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml:5777]: Not sure if it is needed for this PR.

Comment (papdaniel) [reply operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml:5777]: True, it is not needed, I can remove. The `make gen` generated this. Shall I delete it?

Comment (shamsher31) [reply operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml:5777]: Please rebase with master and then run `make gen`. It's not needed so you can remove it.

**Comment by jacob-delgado** [comment]
cc @papdaniel Please run `make gen` and push up your changes.

**Comment by papdaniel** [comment]
> cc @papdaniel Please run `make gen` and push up your changes.

I did `make  gen` then removed the part mentioned above from the result, now some tests fail, obviously. I don't know where is the source of those lines, I didn't make that change. Probably someone earlier forgot to run `make gen`. Now how should I fix that?

**Comment by jacob-delgado** [comment]
Maybe a rebase, then a make gen and push up any changes?

**Comment by papdaniel** [comment]
> Maybe a rebase, then a make gen and push up any changes?

I did that, up above you can see, the guys told me to remove that few lines... Actually I rebased at least five times already, nothing happened. It is here since february. Will you merge it if I rebase it again?

**Comment by papdaniel** [comment]
So, what's going to happen now?

**Comment by jacob-delgado** [comment]
> So, what's going to happen now?

If you rebase it I can approve it. Make sure to tag me once it's been rebased and all tests are passing.

**Comment by papdaniel** [comment]
@jacob-delgado done

**Review by Monkeyanator — COMMENTED** [review]
LGTM! Thanks for your patience getting this through @papdaniel. Yes the operator docs should be updated, we can make sure that happens before 1.12.

**Review by jacob-delgado — APPROVED** [review]

**Review by Monkeyanator — APPROVED** [review]

**Comment by tehlers320** [comment]
Will the chart be released with these updates soon?

**Comment by js-timbirkett** [comment]
💩 💩 💩 💩 💩 💩 💩 💩 💩 💩 💩  - How this got signed off and merged... I'll never know. Thanks for all the breaking changes 🤦
